feat(rtc) : allow custom rtc options

arthur740212 opened the pull request:
This PR creates a `localRTCOptions` variable in the `Lobby` class, that should allow the user to set the options to meet their needs.
The thing I am not so sure about is whether this variable should really exist in this class, or even if it needs to be held beyond the scope of a function/method

arthur740212 committed:
feat(rtc) : allow custom rtc options

paulhazen reviewed (commented):
How do these changes correspond to / overlap with the changes suggested by PR #512?

arthur740212 commented:
> How do these changes correspond to / overlap with the changes suggested by PR #512?

Because a `Lobby` class can be created through the UI scripts, which are modifiable.
By giving `localRTCOptions` a value, the `CreateLobby` method will have the custom local RTC options set.
Otherwise it will use the default that we used to set.

Still, I am missing the Join Lobby part.

paulhazen commented:
> Still, I am missing the Join Lobby part.

Has this PR been updated to include this part?

arthur740212 commented:
> > Still, I am missing the Join Lobby part.
> 
> Has this PR been updated to include this part?

Not yet. I am still trying to figure out how to pass Lobby.localRTCOptions across connection.

arthur740212 committed:
fix(rtc) : Make localRTCOption a member of LobbyManager

The options aren't directly tied to the lobby, instead they are more so game specific.
This change moves the options under lobby manager and could be accessed from user scripts.
Both JoinLobby and CreateLobby will use the custom options if a user sets them manually, otherwise the default options would be used.

arthur740212 commented:
Updated.

virtualboys commented:
Okay yes this solution looks good to me!

paulhazen reviewed (approved):
Looks good to me.
